arkouda/random/_generator.py

```diff
@@ -56,6 +59,20 @@ def __str__(self):
         _str += "(PCG64)"
         return _str
 
+    # chpl has to know the type of the generator, in order to avoid having to declare
+    # the type of the generator beforehand (which is not what numpy does)
+    # we declare a generator for each type and fast-forward the state
+    def _generator_msg(self, dtype):
+        return generic_msg(
+            cmd="createGenerator",
+            args={
+                "dtype": akdtype(dtype).name,
+                "has_seed": self._has_seed,
+                "seed": self._initial_seed,
+                "state": self._initial_state,
+            },
+        )
```

Comment:
I think this creates a new generator every time and adds it to the symbol table. I think right now we declare a generator for each type and fetch the type we need from the symbol table? 

I think this way would work cause we fast forward the state, but it seems to make unnecessary copies

on master:
```python
>>> rng = ak.random.default_rng(11)

# one per type
>>> ak.list_symbol_table()
['id_RjcKYUF_2', 'id_RjcKYUF_3', 'id_RjcKYUF_1', 'id_RjcKYUF_4']

# call integers repeatedly (this shouldn't add any new generators
>>>  pda_names = []
   ...: for i in range(10):
   ...:     pda = rng.integers(20, size=3)
   ...:     pda_names.append(pda.name)
   ...:

# symbol table ignoring the names of pdarray created
>>> set(ak.list_symbol_table()) -  set(pda_names)
{'id_RjcKYUF_1', 'id_RjcKYUF_2', 'id_RjcKYUF_3', 'id_RjcKYUF_4'}
```

with this PR:
```python
>>> rng = ak.random.default_rng(11)

# initally no generators are in symbol table
>>> ak.list_symbol_table()
[]

>>> pda_names = []
   ...: for i in range(10):
   ...:     pda = rng.integers(20, size=3)
   ...:     pda_names.append(pda.name)
   ...:

# symbol table ignoring pda names shows many more generator entries have been added to the sym table
>>> set(ak.list_symbol_table()) -  set(pda_names)
{'id_DYklbur_1',
 'id_DYklbur_11',
 'id_DYklbur_13',
 'id_DYklbur_15',
 'id_DYklbur_17',
 'id_DYklbur_19',
 'id_DYklbur_3',
 'id_DYklbur_5',
 'id_DYklbur_7',
 'id_DYklbur_9'}
```

Reply:
It should be fixed now.  Sorry about that.